helixbass/tsc-rust
# Parse + bind template literal

In this PR:
- parse + bind template literals

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
const a: boolean = `abc${"def"}`;
```
then you should see `post-binding:` debug-logging where the `initializer` of the `VariableDeclaration` is a `TemplateExpression` whose `head` is a `TemplateLiteralLikeNode` of `kind` `TemplateHead` and `text` `"abc"` and `template_spans` contains a `TemplateSpan` whose `expression` is a `StringLiteral` with `text` `"def"` and whose `literal` is a `TemplateLiteralLikeNode` of `kind` `TemplateTail` and `text` `""`, and these should have `parent` bindings initialized

Based on `parse-ambient-declaration`
